safetensors/src/lib.rs

```diff
@@ -96,14 +97,14 @@ fn prepare<'hash, 'data>(
         tensors.push(tensor);
     }
 
-    let metadata: Metadata = Metadata(hmetadata);
+    let metadata: Metadata = Metadata(data_info.clone(), hmetadata);
```

Comment:
New at rust, so not sure if here the best is to explicitly clone or deref with `*` which might move the dictionary?

Reply:
You cannot deref.

More explicitly, if you deref the borrow checker is going to complain.
You have in your possession a `ref` so doing  `*data_info` is OK, but that's the owned value, which you are trying to give to `Metadata`, which this is not yours to give as you don't have ownership ! 

Here we don't really have much of a choice to clone here.
Other option would be to handle the lifetimes everywhere which is too complex for the benefit IMO.